drivers/include/net/netdev/ieee802154.h

```diff
@@ -115,14 +121,65 @@ typedef struct {
     uint8_t page;                           /**< channel page */
     uint16_t flags;                         /**< flags as defined above */
     int16_t txpower;                        /**< tx power in dBm */
+    const netdev_ieee802154_ops_t *ops;     /**< IEEE802.15.4 device ops. Must be set */
     /** @} */
 } netdev_ieee802154_t;
 
+/**
+ * @brief Operations of an IEEEE802.15.4 device
+ */
+struct netdev_ieee802154_ops {
+    /**
+     * @brief Set IEEE addresses to the hardware accelerator of an
+     * network device
+     *
+     * @pre (short_addr != NULL), (ext_addr != NULL)
+     * @note Set to NULL if the device doesn't support address filtering.
+     *
+     * @param[in] dev           network device descriptor
+     * @param[in] short_addr    short address to be set
+     * @param[in] ext_addr      extended address to be set
+     * @param[in] panid         panid to be set
+     */
+    void (*set_hw_addr)(netdev_ieee802154_t *dev, network_uint16_t *short_addr,
+                        eui64_t *ext_addr, uint16_t panid);
+};
+
 /**
  * @brief   Received packet status information for IEEE 802.15.4 radios
  */
 typedef struct netdev_radio_rx_info netdev_ieee802154_rx_info_t;
 
+/**
+ * @brief Check whether a @ref netdev_ieee802154_t device has hardware address
+ *        filter support
+ *
+ * @param[in] dev       network device descriptor
+ *
+ * @return              true if the device supports address filtering
+ */
+static inline bool netdev_ieee802154_has_addr_filter(netdev_ieee802154_t *dev)
+{
+    return dev->ops->set_hw_addr != NULL;
+}
+
+/**
+ * @brief Configure hardware address filter in a given network device
+ *
+ * @pre   @ref dev MUST have hardware address filter! Check with
+ *        @ref netdev_ieee802154_has_addr_filter
+ *
+ * @param[i] dev        network device descriptor
+ * @param[i] short_addr IEEE802.15.4 short address
+ * @param[i] ext_addr   IEEE802.15.4 extended address
+ * @param[i] panid      IEEE802.15.4 PAN-ID
+ */
+static inline void netdev_ieee802154_set_addr_filter(netdev_ieee802154_t *dev, network_uint16_t *short_addr, eui64_t *ext_addr, uint16_t panid)
```

Comment:
Shouldn't both this function and `dev->ops->set_hw_addr()` have an error return in case *something* goes wrong on the device? Or are we expecting everything to work out fine always?

Reply:
This is so low level that if something fails there (wrong SPI transaction, something not initialized) the upper layers have no way to recover. We can add assertions, but checking a general malfunction should be a task of other modules (e.g watchdogs).

Also, none of our functions to set addresses have return codes (e.g https://github.com/RIOT-OS/RIOT/blob/master/drivers/at86rf2xx/at86rf2xx_getset.c#L140)

Reply:
Ok.